### PR TITLE
Translated best practices to French !

### PR DESCRIPTION
**Pull request by GWillmann**
This translation aims to help French developers that are not too familiar with English to be able to apply  Android best practices as well.

### CODE REVIEW

File: translations/French/README.fr.md [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+│  │     │  └─ com/futurice/project
+│  │     ├─ res
+│  │     └─ AndroidManifest.xml
+│  ├─ build.gradle
+│  └─ proguard-rules.pro
+├─ build.gradle
+└─ settings.gradle
+```
+
+La principale différence est que la nouvelle structure sépare explicitement `source sets` (`main`, `androidTest`)), résultant d'un des concepts de Gradle. Vous pourriez par exemple ajouter deux dossiers `payant` `gratuit` dans le dossier `src`qui aura donc le code source de la version payante et de la version gratuite de votre application.
```
Comment (peter-tackage) [inline translations/French/README.fr.md]: Remove extra closing bracket in "(main, androidTest))"

File: translations/French/README.fr.md [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+#### Utiliser Robolectric pour les tests unitaires, Robotium pour les tests graphiques
+#### Utiliser Genymotion comme émulateur
+#### Toujours utiliser Proguard et DexGuard
+
+
+----------
+
+### Android SDK
+
+Mettre le [SDK Android](https://developer.android.com/sdk/installing/index.html?pkg=tools) dans votre répertoire utilisateur (home) ou à un autre endroit indépendant des applications. Certains IDEs inclus le SDK à leur installation, et peuvent le placer dans le même dossier que celui de l'IDE. Cela peut être problématique lorsqu'il faut mettre à jour (ou réinstaller) l'IDE, ou l'orsque l'on change d'IDE. De plus éviter de mettre le SDK dans un dossier système qui pourrait nécessiter des permissions administrateurs si votre IDE n'est pas lancé en tant qu'administrateur.
```
Comment (peter-tackage) [inline translations/French/README.fr.md]: "ou l'orsque" -> "ou lorsque"

File: translations/French/README.fr.md [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+
+L'IDE le plus recommendé en ce moment est : [Android Studio](https://developer.android.com/sdk/installing/studio.html), car il est développé par Google, a bien intégré Gradle, et utilise la nouvelle structure de projet par défaut. De plus il est enfin en version stable et est conçu exprès pour le développement Android.
+
+Vous pouvez utiliser [Eclipse ADT](https://developer.android.com/sdk/installing/index.html?pkg=adt) si vous le souhaitez, mais vous devrez le configurer car il de base il utilise l'ancienne structure de projet et Ant pour builder votre application. Dans ce cas vous devrez utiliser Gradle et `adb` en ligne de commande. Si l'intégration de Gradle dans Eclipse ne marche pas chez vous, vos options sont d'utiliser les lignes de commande juste pour builder ou bien de migrer vers Android Studio. Cette dernière est la meilleure car le plugin ADT est devenu déprécié récemment.
```
Comment (peter-tackage) [inline translations/French/README.fr.md]: "car il de base il utilise" -> "car de base il utilise"

File: translations/French/README.fr.md [hunk 4]
```diff
@@ -0,0 +1,515 @@
...
+
+Si vous n'avez aucune expérience antérieure avec Rx, commencez par mettre en oeuvre cette librairie uniquement sur les réponses des API. Une autre alternative serait de l'utiliser pour les évènements simples liés à l'interface graphique comme les clics ou l'ajout de caractères dans un champ de recherche. Si vous êtes confiants à propos de vos compétences en Rx et que vous souhaitez l'appliquer à l'architecture de votre projet, n'oubliez pas d'écrire des Javadocs au niveau de toutes les parties difficiles. Gardez en tête qu'un autre développeur qui ne connait pas RxJava aura probablement beaucoup de mal à maintenir votre projet. faites de votre mieux pour les aider à comprendre votre code et Rx.
```
Comment (peter-tackage) [inline translations/French/README.fr.md]: Capitalize "faites" in "faites de votre mieux"

File: translations/French/README.fr.md [hunk 5]
```diff
@@ -0,0 +1,515 @@
...
+
+- Toute interface qui comporte uniquement une méthode est compatible lambda et peut être réduite dans la syntaxe la plus compacte.
+- Si vous avez des doutes concernant les paramètres ou autre, écrivez une classe interne normale et laissez Android Studio la compacter en un lambda pour vous.
+
+**Faites attention à la limiation des méthodes dex et évitez d'utiliser beaucoup de librairies.** Lorsque les applications Android sont packagées en un fichier Dex, elles ont une limitation de 65536 méthodes référencées [[1]](https://medium.com/@rotxed/dex-skys-the-limit-no-65k-methods-is-28e6cb40cf71) [[2]](http://blog.persistent.info/2014/05/per-package-method-counts-for-androids.html) [[3]](http://jakewharton.com/play-services-is-a-monolith/). Vous verrez une erreur fatale si jamais vous dépassez cette limie. Pour cette raison utilisez un minimum de librairies et utilisez l'outil [dex-method-counts](https://github.com/mihaip/dex-method-counts) pour déterminer quelles librairies peuvent être utilisées pour rester en dessous de cette limite. Evitez d'utiliser la librairie Guava car elle contient plus de 13k méthodes.
```
Comment (peter-tackage) [inline translations/French/README.fr.md]: "dépassez cette limie" -> "dépassez cette limite"

File: translations/French/README.fr.md [hunk 6]
```diff
@@ -0,0 +1,515 @@
...
+├─ utils
+├─ fragments
+└─ views
+   ├─ adapters
+   ├─ actionbar
+   ├─ widgets
+   └─ notifications
+```
+
+### Ressouces
```
Comment (peter-tackage) [inline translations/French/README.fr.md]: "Ressouces" -> "Ressources"

File: translations/French/README.fr.md [hunk 7]
```diff
@@ -0,0 +1,515 @@
...
+
+**Règles de nommage.** Suivez les conventions en préfixant vos noms du type comme `type_foo_bar.xml`. Exemples: `fragment_contact_details.xml`, `view_primary_button.xml`, `activity_main.xml`.
+
+**Organisation des fichiers XMLs.** Si vous n'êtes pas sûr(e) de l'organisation de vos fichiers XMLs, les conventions suivantes pourraient vous aider :
+
+- Un attribut par ligne, indenté de espaces
```
Comment (peter-tackage) [inline translations/French/README.fr.md]: "de espaces" -> "d'espaces"

File: translations/French/README.fr.md [hunk 8]
```diff
@@ -0,0 +1,515 @@
...
+- `android:id` toujours en tant que premier attribut
+- les `android:layout_****` toujours en haut après `android:id`
+- l'attribut `style` tout en bas
+- le tag `/>` de fermeture soit être sur sa propre ligne pour faciliter l'ajout d'attributs et l'ordonnancement.
```
Comment (peter-tackage) [inline translations/French/README.fr.md]: "fermeture soit être" -> "fermeture doit être"

File: translations/French/README.fr.md [hunk 9]
```diff
@@ -0,0 +1,515 @@
...
+</LinearLayout>
+```
+
+En règle générale, les attributs `android:layout_****` doivent être placés dans les fichiers layouts XML et les autres attributs `android:****` dans les fichiers de style XML. Cette règle a des exceptions mais s'applique bien la plupart du temps. L'idée est de garder uniquement les attributs de contenu et de layout (positionnement, marges, tailles) dans les fichiers de layout XML et de mettre tous les autres liés aux détails de l'apparence (couleurs, padding, police) dans des fichiers de style.
+
+Les exceptions sont :
+
+- `android:id` doit évidement se trouver dans les fichiers layout XML
+- l'attribut `android:orientation` pour un `LinearLayout` a normalement plus de sens dans les fichiers layout XML
+- l'attribut `android:text` devrait être dans les fichiers layout XML car il défini du contenu
```
Comment (peter-tackage) [inline translations/French/README.fr.md]: "car il défini" -> "car il définit"

File: translations/French/README.fr.md [hunk 10]
```diff
@@ -0,0 +1,515 @@
...
+
+### Emulateurs
+
+Si votre métier est de développer des applications Android, achetez une license de [l'émulateur Genymotion](http://www.genymotion.com/). Les émulateurs Genymotion ont un taux de rafraichissement plus rapide que les machines virtuelles AVD. Ils ont aussi des outils qui permettent de faire des démos de votre application, de simuler tel ou tel type de connection internet, de simuler des positions GPS, etc. Ils sont aussi idéals pour faire des tests connectés. Vous aurez accès à beaucoup de différents périphériques (pas tous mais beaucoup) donc le coup d'achat d'une license de Genymotion revient moins cher qu'acheter le nombre équivalent de périphériques.
+
+Les inconvénients sont : les émulateurs Genymotion ne contiennent pas tous les services Google comme Google Play Store ou Maps. Vous aurez aussi probablement besoin de tester des APIs spécifiques aux téléphones samsung donc vous aurez besoin d'acheter un vrai téléphone pour ça.e Play Store and Maps. You might also need to test Samsung-specific APIs, so it's necessary to have a real Samsung device.
```
Comment (peter-tackage) [inline translations/French/README.fr.md]: Remove the English bit at the end of the paragraph.

File: translations/French/README.fr.md [hunk 11]
```diff
@@ -0,0 +1,515 @@
...
+    }
+    release {
+        signingConfig signingConfigs.release
+        minifyEnabled true
+        proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+    }
+}
+```
+
+Pour déterminer quels bouts de code doit être gardé tel quel et lesquels doivent ête supprimés ou obfusqués, vous devez spécifier un ou plusieurs points d'entrée dans votre code.
```
Comment (peter-tackage) [inline translations/French/README.fr.md]: "doit être gardé" -> "doivent être gardés"

**Comment by peter-tackage** [comment]
Thanks for the translation @GWillmann! If you could fix the minor issues (mostly typos) I've noted, then squash the commits and then it will then be good to merge.
cheers.

**Comment by GWillmann** [comment]
Sorry for the delay @peter-tackage !
I have fixed all the typos, it should be good to go.
Cheers.

**Pull request merged**